components/inbound-endpoints/org.wso2.carbon.inbound.endpoint.persistence/src/main/java/org/wso2/carbon/inbound/endpoint/persistence/InboundEndpointsDataStore.java

```diff
@@ -47,20 +48,22 @@
     private Map<String,Set<String>> endpointPollingInfo;
     private Registry registry = null;
     private final String rootPath = RegistryResources.ROOT + "esb/inbound/inbound-endpoints/";
-
+    public static final String STORE_ARTIFACTS_IN_REGISTRY = "synapse.artifacts.registry.storage.enabled";
     private static InboundEndpointsDataStore instance = new InboundEndpointsDataStore();
 
     public static InboundEndpointsDataStore getInstance() {
         return instance;
     }
+    private boolean saveInboundEndpointInRegistry =
+            SynapsePropertiesLoader.getBooleanProperty(STORE_ARTIFACTS_IN_REGISTRY, false);
 
     private InboundEndpointsDataStore() {
         try {
             registry = ServiceReferenceHolder.getInstance().getRegistry();
         } catch (RegistryException e) {
             handleException("Error while obtaining a registry instance", e);
         }
-        if(!Boolean.parseBoolean(System.getProperty("NonRegistryMode"))) {
+        if (!Boolean.parseBoolean(System.getProperty("NonRegistryMode")) || saveInboundEndpointInRegistry) {
```

Comment:
Why are we introducing another property which almost is the same as 'NonRegistryMode'?

Reply:
If we use nonRegistrymode in APIM it will not be able to use since we have usage of registry